runtime/ioext/ioext.go

```diff
@@ -0,0 +1,28 @@
+package ioext
+
+import (
+	"io"
+)
+
+// ReadSeekCloser implements io.Reader, io.Seeker, and io.Closer. It is trivially implemented by os.File.
+type ReadSeekCloser interface {
+	io.Reader
+	io.Seeker
+	io.Closer
+}
+
+// NopCloser is useful in testing where something that implements ReadSeekCloser is needed
+// If something that implements io.ReadSeeker is passed in, it will give it a noop close function.
+func NopCloser(r io.Reader) ReadSeekNopCloser {
+	return ReadSeekNopCloser{r.(io.ReadSeeker)}
+}
+
+// ReadSeekNopCloser is an implementation of ReadSeekCloser that wraps io.ReadSeekers
+type ReadSeekNopCloser struct {
```

Comment:
You don't need to export this type...

You create it with `NopCloser` whose return type should `ReadSeekCloser`